_pages/documentation/gem5-stdlib/8-extending-disk-images.md

```diff
@@ -0,0 +1,244 @@
+   packer init /path/to/packer/script
+   packer build /path/to/packer/script
+   ```
+
+**Note:** If you want to view the terminal of the image during the build process, you can use a VNC viewer. Packer will provide a VNC port during the build, which you can connect to using a VNC viewer.
+
+## Important Notes
+
+- If you are extending the RISC-V base image, please note that you would need to remount the file system as read/write upon logging in before enabling network.
+The reason we need to do this is because qemu cant not handle `m5 exit` during the boot process properly so it boots into read only for safety.
+The boot command will look something like this:
+
+```hcl
+ boot_command =["<wait45>",
+                "gem5<enter><wait>",
+                "12345<enter><wait>",
+                "sudo mount -o remount,rw /<enter><wait>", // remounting file system
+                "12345<enter><wait>",
+                "sudo mv /etc/netplan/50-cloud-init.yaml.bak /etc/netplan/50-cloud-init.yaml<enter><wait>",
+                "sudo netplan apply<enter><wait>",
+                "<wait>"
+              ]
+```
+
+## Running disk images in gem5
+
+As these disk iamges throw m5 exit events, we need to handle those to succesfully run the image to boot.
+
+## What are Exit Events?
+
+In gem5, exit events are a way to exit the simulation loop. Exit events can be used to perform operations outside the simulation when triggered, such as resetting stats, switching CPU types, dumping stats, taking checkpoints, etc. By default, the simulation will exit when an exit event is triggered, but we can modify this behavior by writing our own exit event handlers.
+
+## Writing Custom Exit Event Handlers
+
+gem5 expects exit event handlers to be generators, and we can override them by passing our custom generator to the `simulator` object. Let’s say we have a resource that throws an `m5 exit` at some point during execution. Instead of stopping and exiting the simulation, we want to switch processors (assuming we are using the `SimpleSwitchableProcessor`). To do this, we would need to import `ExitEvent` from `gem5.simulate.exit_event`:
+
+```python
+from gem5.simulate.exit_event import ExitEvent
+```
+
+Now, let's write our handler:
+
+```python
+def exit_event_handler():
+    # Switch the CPU type
+    print("Switching CPU types")
+    processor.switch()
+    yield False  # Continue the simulation
+```
+
+This handler will print a message and switch CPU types. We yield `False` because we want the simulation to continue.
+
+To add our handler to the `simulator` object, we do the following:
+
+```python
+simulator = Simulator(
+    board=board,
+    on_exit_event={
+        # Override the default behavior for the first m5 exit event
+        ExitEvent.EXIT: exit_event_handler()
+    },
+)
+```
+
+**NOTE:** We pass a call to our handler (`exit_event_handler()`) and not the handler itself (`exit_event_handler`) to the simulator.
+
+## Exit Events Thrown by the New Disk Images
```

Comment:
Ok, can we just put a hold on this PR until I finish with my exit events and see if anything in here needs changed? There's a lot here and I I don't want to waste our time tweeking it if we need to change it again in two weeks